packages/ethereum-wallets/README.md

```diff
@@ -29,42 +29,112 @@ yarn add near-api-js @web3modal/wagmi wagmi viem @tanstack/react-query @near-wal
 ```
 
-Then use it in your dApp:
+Then use it in your App:
 
 Visit https://docs.walletconnect.com for the latest configuration of Web3Modal.
 
+Tested versions from `/examples`:
+```json
+"dependencies": {
+  "@web3modal/wagmi": "5.0.6",
+  "@tanstack/react-query": "5.24.8",
+  "viem": "2.16.2",
+  "wagmi": "2.10.9",
+}
+```
+
 ```ts
 import type { Config } from "@wagmi/core";
+import type { Chain } from "@wagmi/core/chains";
 import { reconnect, http, createConfig } from "@wagmi/core";
 import { walletConnect, injected } from "@wagmi/connectors";
 import { setupWalletSelector } from "@near-wallet-selector/core";
 import { setupEthereumWallets } from "@near-wallet-selector/ethereum-wallets";
 
+// Mainnet
+const near: Chain = {
+  id: 397,
+  name: "NEAR Protocol",
+  nativeCurrency: {
+    decimals: 18,
+    name: "NEAR",
+    symbol: "NEAR",
+  },
+  rpcUrls: {
+    default: { http: ["https://eth-rpc.mainnet.near.org"] },
+    public: { http: ["https://eth-rpc.mainnet.near.org"] },
+  },
+  blockExplorers: {
+    default: {
+      name: "NEAR Explorer",
+      url: "https://eth-explorer.near.org",
+    },
+  },
+}
+
+// Testnet
+/*
+const near: Chain = {
+  id: 398,
+  name: "NEAR Protocol Testnet",
+  nativeCurrency: {
+    decimals: 18,
+    name: "NEAR",
+    symbol: "NEAR",
+  },
+  rpcUrls: {
+    default: { http: ["https://eth-rpc.testnet.near.org"] },
+    public: { http: ["https://eth-rpc.testnet.near.org"] },
+  },
+  blockExplorers: {
+    default: {
+      name: "NEAR Explorer",
+      url: "https://eth-explorer-testnet.near.org",
+    },
+  },
+  testnet: true,
+};
+*/
+
+// Get a project ID at https://cloud.walletconnect.com
+const projectId = ""
+
 const wagmiConfig: Config = createConfig({
   chains: [near],
   transports: {
     [near.id]: http(),
   },
   connectors: [
-    walletConnect({ projectId, metadata, showQrModal: false }),
+    walletConnect({
+      projectId,
+      metadata: {
+        name: "NEAR Guest Book",
+        description: "A guest book with comments stored on the NEAR blockchain",
+        url: "https://near.github.io/wallet-selector",
+        icons: ["https://near.github.io/wallet-selector/favicon.ico"],
+      },
+      showQrModal: false
+    }),
     injected({ shimDisconnect: true }),
   ],
 });
-reconnect(wagmiConfig);
 
 const web3Modal = createWeb3Modal({
-  wagmiConfig: config,
-  // Get a project ID at https://cloud.walletconnect.com
+  wagmiConfig,
   projectId,
 });
 
-const _selector = await setupWalletSelector({
-  network: "mainnet",
-  debug: true,
-  modules: [
-    setupEthereumWallets({ wagmiConfig, web3Modal }),
-  ],
-});
+export const WalletSelectorContextProvider = () => {
+  const init = useCallback(async () => {
+    const _selector = await setupWalletSelector({
+      network: "mainnet",
+      modules: [
+        setupEthereumWallets({ wagmiConfig, web3Modal }),
+      ],
+    });
+    const _modal = setupModal(_selector, { contractId: "" })
```

Comment:
@paouvrard what is this `setupModal` function? I don't see it imported.

Reply:
https://github.com/near/wallet-selector/pull/1189/commits/bc3b184366439a3935881448cf48ef837e1e0a69